azure-app-service/massdriver.yaml

```diff
@@ -0,0 +1,195 @@
+        maximum_worker_count: 10
+  required:
+    - application
+    - repo
+    - dns
+  properties:
+    application: 
+      title: Application
+      description: Application configuration
+      type: object
+      required:
+        - location
+        - sku_name
+        - minimum_worker_count
+        - maximum_worker_count
+      properties:
+        location:
+          title: Location
+          description: Azure location
+          type: string
+          $md.immutable: true
+          oneOf:
+            - title: West US
+              const: westus
+            - title: West US 2
+              const: westus2
+            - title: West US 3
+              const: westus3
+            - title: East US
+              const: eastus
+            - title: East US 2
+              const: eastus2
+            - title: North Central US
+              const: northcentralus
+            - title: South Central US
+              const: southcentralus
+            - title: Central US
+              const: centralus
+        sku_name:
+          title: Size
+          description: Azure App Service Plan size
+          type: string
+          oneOf:
+            - title: 1 vCPU, 1.75GB RAM
+              const: S1
+            - title: 2 vCPU 8GB RAM
+              const: P1v3
+            - title: 4 vCPU 16GB RAM
+              const: P2v3
+            - title: 8 vCPU 32GB RAM
+              const: P3v3
+        minimum_worker_count:
+          title: Minimum instances
+          description: Minimum number of instances for autoscale.
+          type: integer
+          minimum: 1
+          maximum: 30
+        maximum_worker_count:
+          title: Maximum instances
+          description: Maximum number of instances for autoscale.
+          type: integer
+          minimum: 1
+          maximum: 30
+        notification_email:
+          title: Notification email
+          description: Receive alerts regarding autoscale events.
+          type: string
+          format: email
+          example: user@example.com
+    repo:
+      title: Repository
+      description: Repository configuration
+      type: object
+      required:
+        - docker_image
+        - docker_image_tag
+      properties:
+        docker_image:
+          title: Docker image
+          description: "Docker image source. Example: docker.io/library/nginx"
+          type: string
+        docker_image_tag:
+          title: Docker image tag
+          description: Docker image tag name
+          type: string
+    dns:
+      title: DNS
+      description: DNS configuration
+      type: object
+      required:
+        - enable_dns
+      properties:
+        enable_dns:
+          title: Enable DNS
+          description: Enable DNS configuration for the application.
+          type: boolean
+      dependencies:
+        enable_dns:
+          oneOf:
+            - properties:
+                enable_dns:
+                  const: true
+                zone_name:
```

Comment:
Can we simplify these settings at all?  People who don't know DNS much will simply know they want their app on a certain URL. They may not understand the different between TXT records and CNAME records.  We can make the CNAME and generate a TXT record for validation for them.

Reply:
Yea we can, I'll work on this tonight